src/kakarot/library.cairo

```diff
@@ -185,21 +185,16 @@ namespace Kakarot {
         pedersen_ptr: HashBuiltin*,
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
-    }(bytecode_len: felt, bytecode: felt*) -> (
+    }(nonce: felt, bytecode_len: felt, bytecode: felt*) -> (
```

Comment:
why do you need to get the `nonce` as input here instead of retrieving it like in `CREATE`?

Reply:
hmmm true, I guess i can also fetch the transaction info here.
No unnecessary passing of arguments.
will fix!